interfaces/GithubFactory.py

```diff
@@ -216,19 +217,31 @@ def find_repos(self, stars: int, lang: str, nb_repo: int):
             if i == int(nb_repo):
                 break
     
-    def create_test_repo(self, shaBase, repoFullName: str, path_repos: str):
-        """Creates a test repository by cloning the repository specified by `self.repoFullName` and checking out the base commit specified by `shaBase`.
-        
-        If the test repository directory does not exist, it will be created under the `./test` directory. The repository will then be cloned from the specified URL and the base commit will be checked out.
+    def setup_repo_and_get_file_diff(self, shaBase, repoFullName: str, path_repos: str):
```

Comment:
If there is a and in the function name, it means that the function has to be split in 2 functions, because function need to have a single purpose for better comprehension